core/src/utils/directive.ts

```diff
@@ -227,76 +228,150 @@ export const mergeDirectives =
-	 */
-	classNames?: Record<string, boolean | ReadableSignal<boolean>>;
-}
+			for (const [attributeName, value] of Object.entries(attributes ?? {})) {
+				if (value != null) {
+					unsubscribers.push(bindAttribute(node, attributeName, toReadableStore(value)));
+				}
+			}
+
+			for (const [styleName, value] of Object.entries(styles ?? {})) {
+				if (value) {
+					unsubscribers.push(bindStyle(node, styleName, toReadableStore(value)));
+				}
+			}
+
+			for (const [className, value] of Object.entries(classNames ?? {})) {
+				unsubscribers.push(bindClassName(node, className, toReadableStore(value)));
+			}
+
+			return {
+				update: (args: T) => args$.set(args),
+				destroy: () => unsubscribers.forEach((fn) => fn()),
+			};
+		},
+		{propsFn},
+	);
+
+	return directive;
+};
 
 /**
- * Creates a directive for server-side rendering with bindable elements.
- * This directive binds events, attributes, styles, and classNames to an HTML element.
+ * Creates a directive for adding specified CSS class names to an HTML element.
  *
- * @param propsFn - A function that returns the AttributesDirectiveProps with the data to bind.
- * This function can take an optional parameter that corrspond to the second parameter of the created directive.
- * @returns A directive object with bound events, attributes, styles, and classNames.
+ * @param classNames - An array of CSS class names to be added.
+ * @returns A directive object with bound class names.
  */
-export const createAttributesDirective =
-	<T = void>(propsFn: (arg: ReadableSignal<T>) => AttributesDirectiveProps) =>
-	(node: HTMLElement, args: T) => {
-		const unsubscribers: (() => void)[] = [];
-		const args$ = writable(args);
+export function createClassDirective(...classNames: string[]) {
+	const classNamesObj: AttributesDirectiveProps['classNames'] = {};
+	for (const className of classNames) {
+		classNamesObj[className] = true;
+	}
+	return createAttributesDirective(() => ({classNames: classNamesObj}));
+}
+
+export type AttributeDirectiveAndParam<T> = [AttributeDirective<T>, T];
 
-		const {events, attributes, styles, classNames} = propsFn(args$);
+/**
+ * Returns an object with the attributes, style and class keys containing information derived from a list of directives.
+ *
+ *   - The `attributes` value is a JSON representation of key/value attributes, excepted for the `class` and `style` attributes
+ *   - The `classNames` value is an array of string representing the classes to be applied
+ *   - The `style` value is a JSON representation of the styles to be applied
+ *
+ * @param directives - List of directives to generate attributes from. Each parameter can be the directive or an array with the directive and its parameter
+ * @returns JSON object with the `attributes`, `class` and `style` keys.
+ */
+export function ssrAttributesData<T extends any[]>(...directives: {[K in keyof T]: AttributeDirectiveAndParam<T[K]> | AttributeDirective<void>}) {
```

Comment:
Actually, even if this function is mostly intended for SSR, nothing prevents it from being used for something else.
Maybe we should reserve the `ssr` prefix for functions that just call the unprefixed function of the same name only when not in a browser environment.